pkg/notify/mattermost.go

```diff
@@ -98,11 +98,19 @@ func (m *Mattermost) SendEvent(event events.Event) error {
 			log.Error("Failed to send message. Error: ", resp.Error)
 			// send error message to default channel
 			msg := fmt.Sprintf("Unable to send message to Channel `%s`: `%s`\n```add Botkube app to the Channel %s\nMissed events follows below:```", event.Channel, resp.Error, event.Channel)
-			go m.SendMessage(msg)
+			err := m.SendMessage(msg)
+			if err != nil {
+				// TODO: Append error and return them all at once
+				log.Errorf("while sending message to default channel: %s", err.Error())
+			}
 			// sending missed event to default channel
 			// reset event.Channel and send event
 			event.Channel = ""
-			go m.SendEvent(event)
+			err = m.SendEvent(event)
+			if err != nil {
+				// TODO: Append error and return them all at once
+				log.Errorf("while sending event to default channel: %s", err.Error())
```

Comment:
Should we return error?

Reply:
Agree 👍  I will use https://github.com/hashicorp/go-multierror to append errors, as in this case I only returned the original error 🙂